ui/components/app/alert-system/contexts/alertMetricsContext.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { createContext, useContext, ReactNode, useMemo } from 'react';
+import {
+  UseAlertSystemMetricsProps,
+  useConfirmationAlertMetrics,
+} from '../../../../pages/confirmations/hooks/useConfirmationAlertMetrics';
+
+const AlertMetricsContext = createContext<{
+  trackAlertMetrics: (props?: UseAlertSystemMetricsProps) => void;
+} | null>(null);
+
+export const AlertMetricsProvider: React.FC<{ children: ReactNode }> = ({
+  children,
+}) => {
+  const { trackAlertMetrics } = useConfirmationAlertMetrics();
+
+  const value = useMemo(() => ({ trackAlertMetrics }), [trackAlertMetrics]);
+
+  return (
+    <AlertMetricsContext.Provider value={value}>
+      {children}
+    </AlertMetricsContext.Provider>
+  );
+};
```

Comment:
We use context for shared state, but there is no shared state. Can we avoid creating this new context.

Reply:
I apologize for any confusion. I have now updated the PR description to explain the rationale behind this approach. The context is necessary to aggregate and persist state across multiple events (e.g., user clicks on inline alerts, visualizations, and action clicks) within the alert system. This ensures that all interactions are captured and combined into a single set of properties, which are then added to the transaction event. Without the context, we would not be able to maintain this aggregated state across different components effectively.

Reply:
Since hook `useConfirmationAlertMetrics` is not used outside the context, may be code can be embedded in context itself.